feature/interface/holdtime/otg_tests/holdtime_test/holddown_timers_test.go

```diff
@@ -0,0 +1,670 @@
+		otgStateChangeTsStr = time.Now().UTC().Format(time.RFC3339Nano)
+
+		ate.OTG().SetControlState(t, portStateAction)
+
+		// TC2 Step 3
+		t.Log("Step 3 sleeping 1000ms")
+		time.Sleep(1000 * time.Millisecond)
+	}
+
+	// Step 4. Read timestamp of last oper-status change  form DUT port-1 (DUT_LAST_CHANGE_TS)
+	postStateTSSTR := gnmi.Get(t, dut, gnmi.OC().Interface(aggID).LastChange().State())
+	DutLastChangeTS2STR := time.Unix(0, int64(postStateTSSTR)).UTC().Format(time.RFC3339Nano)
+
+	DutLastChangeOper2 := gnmi.Get(t, dut, gnmi.OC().Interface(aggID).OperStatus().State())
+
+	var expectedStatus oc.E_Interface_OperStatus
+	if actionState == "UP" {
+		expectedStatus = oc.Interface_OperStatus_UP
+	} else if actionState == "DOWN" {
+		expectedStatus = oc.Interface_OperStatus_DOWN
+	}
+
+	// Step 5. verify oper-status is DOWN
+	if DutLastChangeOper2 != expectedStatus {
+		t.Errorf("Interface %s status got %v, want %v", aggID, DutLastChangeTS2STR, expectedStatus.String())
+	} else {
+		t.Logf("Interface %s status got %v, want %v", aggID, DutLastChangeTS2STR, expectedStatus.String())
+	}
+
+	// convert string type change to time.time
+	otgStateChangeTs, err := time.Parse(time.RFC3339Nano, otgStateChangeTsStr)
+	DutLastChangeTS2, err := time.Parse(time.RFC3339Nano, DutLastChangeTS2STR)
+
+	if err != nil {
+		t.Fatalf("failed to parse event timestamp: %v", err)
+	}
+
+	// Step 6. verify oper-status last change time has changed
+	t.Log("Compare if pre and post timestamps are the same for the last change before and after shut event")
+	if DutLastChangeTS1 == DutLastChangeTS2STR {
+		t.Fatalf("Before Trigger Last Change was %v after trigger Last Change was %v", DutLastChangeTS1, DutLastChangeTS2STR)
+	} else {
+		t.Logf("Before Trigger Last Change was %v after trigger Last Change was %v", DutLastChangeTS1, DutLastChangeTS2STR)
+	}
+
+	// convert to time objects
+	otgStateChangeTs = otgStateChangeTs.UTC()
+	DutLastChangeTS2 = DutLastChangeTS2.UTC()
+
+	return otgStateChangeTs, DutLastChangeTS2, expectedStatus.String(), DutLastChangeOper2.String()
+
+}
+
+// verifyPortsUp asserts that each port on the device is operating.
+func verifyPortsStatus(t *testing.T, dut *ondatra.DUTDevice, portState string) {
+	t.Helper()
+
+	t.Logf("Checking Oper Status on %s", aggID)
+	status := gnmi.Get(t, dut, gnmi.OC().Interface(aggID).OperStatus().State())
+
+	// Determine the expected status based on the portState argument.
+	var want oc.E_Interface_OperStatus
+	if portState == "UP" {
+		want = oc.Interface_OperStatus_UP
+	} else {
+		want = oc.Interface_OperStatus_DOWN
+	}
+
+	// check the status and log the result.
+	if status != want {
+		t.Fatalf("Failed: %s Status: got %v, want %v", aggID, status, want)
+	} else {
+		t.Logf("Pass: %s Status: got %v, want %v", aggID, status, want)
+	}
+}
+
+func TestHoldTimeConfig(t *testing.T) {
+
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+
+	t.Run(fmt.Sprintf("Configure NTP Server on DUT"), func(t *testing.T) {
```

Comment:
Here and throughout the rest of the code, please fix static check `unnecessary use of fmt.Sprintf (S1039)`

```suggestion
	t.Run("RT-5.5.1: Configure NTP Server on DUT", func(t *testing.T) {
```